lib/utils_input.go

```diff
@@ -0,0 +1,129 @@
+package lib
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+)
+
+type INPUT_TYPE string
+
+const (
+	INPUT_TYPE_IP       INPUT_TYPE = "IP"
+	INPUT_TYPE_IP_RANGE INPUT_TYPE = "IPRange"
+	INPUT_TYPE_CIDR     INPUT_TYPE = "CIDR"
+	INPUT_TYPE_FILE     INPUT_TYPE = "File"
+	INPUT_TYPE_ASN      INPUT_TYPE = "ASN"
+	INPUT_TYPE_UNKNOWN  INPUT_TYPE = "Unknown"
+)
+
+// getInputFrom retrieves input data from various sources and processes it using the provided operation.
+// The operation is called for each input string with input type.
+//
+//	Usage:
+//	err := getInputFrom(inputs,
+//		true,
+//		func(input string, inputType INPUT_TYPE) error {
+//			switch inputType {
+//			case INPUT_TYPE_IP:
+//				// Process IP here
+//			default:
+//				return ErrNotIP
+//			}
+//			return nil
+//		})
+func getInputFrom(
+	inputs []string,
+	stdin bool,
+	op func(input string, inputType INPUT_TYPE) error,
+) error {
+	if !stdin && len(inputs) == 0 {
+		return nil
+	}
+
+	// start with stdin.
+	if stdin {
+		stat, _ := os.Stdin.Stat()
+
+		isPiped := (stat.Mode() & os.ModeNamedPipe) != 0
+		isTyping := (stat.Mode()&os.ModeCharDevice) != 0 && len(inputs) == 0
+
+		if isTyping {
+			fmt.Println("** manual input mode **")
+			fmt.Println("one input per line:")
+		}
+
+		if isPiped || isTyping || stat.Size() > 0 {
+			inputs = append(inputs, ReadStringsFromStdin()...)
```

Comment:
This should actually be doing an `op` per string coming from `stdin` on the fly, instead of having it be returned. Otherwise for operations like `prips` this won't scale, because it'll store the whole thing in memory before proceeding and in some inputs that won't be even possible (huge IP ranges/cidrs).